# What is the First Day of the Week?

Whether the Gregorian calendar shows Sunday or Monday as the first day of the week depends on where you live.

Do more countries start the week on Sunday or Monday? What about people? What about by continent?

The file `first-day-of-week.csv` shows the first day of the week for each territory. The file `population.csv` shows the population in the year 2020 for each territory in millions, and the file `four-regions.csv` specifies whether each territory is in asia, europe, africa, or the americas. 


**NOTE: This project originated from an open-ended project in Cisco Networking Academy's *Data Science Essentials with Python: Coding for Answer Chapter* course. I used the provided project as a starting point and independently extended the analysis.**

In [13]:
import pandas as pd
df = pd.read_csv('first-day-of-week.csv')
df.head()

,territory,alpha3,first_day,units,paper
0,Afghanistan,AFG,sat,metric,A4
1,Aland Islands,ALA,mon,metric,A4
2,Albania,ALB,mon,metric,A4
3,Algeria,DZA,sat,metric,A4
4,American Samoa,ASM,sun,metric,A4


In [14]:
pop = pd.read_csv('population.csv')
pop.head()

,alpha3,population
0,AFG,39.07
1,ALB,2.87
2,DZA,44.04
3,AND,0.08
4,AGO,33.45


In [15]:
regions = pd.read_csv('four-regions.csv')
regions.head()

,alpha3,four_regions
0,AUS,asia
1,BRN,asia
2,KHM,asia
3,CHN,asia
4,FJI,asia


### Project Ideas

- How many territories show Friday, Saturday, Sunday, and Monday as the `first_day` of the week?

- How many people start the week on Friday, Saturday, Sunday, and Monday?

- Which of the `four_regions` predominantly start the week on Sunday? On Monday? Are there any regions that are more divided between Sunday and Monday?

**QUESTION 1: How many territories show Friday, Saturday, Sunday, and Monday as the `first_day` of the week?**

In [16]:
df.groupby('first_day').size()


first_day
fri      1
mon    186
sat     15
sun     55
dtype: int64

**ANSWER:** The results show that Monday is the most as the first day of the week, with a rcord of 186 terrorities using it. Followed by Sunday, with 55 territories, Saturday with 15 territories, and lastly by Friday with only one sole territory using it as first day of the week.

**QUESTION 2:  How many people start the week on Friday, Saturday, Sunday, and Monday?**

In [17]:
merged = df.merge(pop, on = 'alpha3', how = 'left')
pop_by_fday = merged.groupby('first_day')['population'].sum()
pop_by_fday

first_day
fri       0.50
mon    3600.19
sat     431.79
sun    3844.08
Name: population, dtype: float64

**ANSWER:** Majority of the population use Sunday as the first day of the week, recording 3844.08 million people. It is then followed by Monday with 3600.19 million people, Saturday with 431.79 million people. Friday is used by the least portion of the population, recording 0.50 million people.

**QUESTION 3: Which of the `four_regions` predominantly start the week on Sunday? On Monday? Are there any regions that are more divided between Sunday and Monday?**

**3.1 Region that predominantly start the week on Sunday:**

In [27]:
region = df.merge(regions, on = 'alpha3', how = 'left')
sun = region.query('first_day == "sun"').groupby('four_regions').size().reset_index(name='sun_count')
sun

,four_regions,sun_count
0,africa,6
1,americas,20
2,asia,21
3,europe,2


**ANSWER:** Asia seem to predominantly start the week on Sunday since it has the most territories that use Sunday compared to other regions.

**3.2 Region that predominantly start the week on Monday:**

In [26]:
mon = region.query('first_day == "mon"').groupby('four_regions').size().reset_index(name='mon_count')
mon

,four_regions,mon_count
0,africa,43
1,americas,15
2,asia,27
3,europe,46


**ANSWER:** Europe use Monday predominantly start their week on Monday compared to the other regions, with a total of 46 territories using it.

**3.3 Are there any regions that are more divided between Sunday and Monday?**

In [ ]:
# merging the two dataframes
sunmon = sun.merge(mon, on = 'four_regions', how = 'left')
sunmon

,four_regions,sun_count,mon_count
0,africa,6,43
1,americas,20,15
2,asia,21,27
3,europe,2,46


In [42]:
#solving for the ratio of sun and mon first day of week for each region
sunmon['ratio'] = (sunmon['sun_count']/sunmon['mon_count'])

#selecting which ratio is closer to 1 by subtravting the ratio from 1 and taking the absolute value
#the value closer to zero is the one that is closer to 1
sunmon['closer_ratio'] = abs(sunmon['ratio']-1)
sunmon

,four_regions,sun_count,mon_count,ratio,closer_ratio
0,africa,6,43,0.139535,0.860465
1,americas,20,15,1.333333,0.333333
2,asia,21,27,0.777778,0.222222
3,europe,2,46,0.043478,0.956522


**ANSWER:** Asia is the most divided region between Sunday and Monday as the first day of the week compared to other regions.